src/parser/expression/regexes.ts

```
@@ -54,33 +147,64 @@ export class NumberRegex extends ExpressionRegex<number> {
     }
 
     public getRegex(index: number) {
-        return `(?<number${index}>\\$?\\d+(\\.\\d+)?)`
+        return `\\b(?<number${index}>\\d+(\\.\\d+)?)\\b`
     }
 
     public getValue(str: string): number {
-        if (str.startsWith(`$`)) {
-            return Number.parseInt(str.replace(`$`, ``), 10)
-        }
+        return Number(str)
+    }
+}
+
+export class DateRegex extends ExpressionRegex<Date> {
+    public constructor() {
+        super({
+            keyword: `{date}`,
+            groupName: `date`,
+            keywordRegex: /{date}/g,
+        })
+    }
+
+    public getRegex(index: number) {
+        const dateRegex = `[0-9]{1,2}\/[0-9]{1,2}\/[0-9]{4}(?: [0-9]{2}:[0-9]{2}:[0-9]{2})?`
+        const isoDateRegex = `(?:\\d{4})-(?:\\d{2})-(?:\\d{2})`
+        const isoDatetimeRegex = `(?:\\d{4})-(?:\\d{2})-(?:\\d{2})T(?:\\d{2}):(?:\\d{2}):(?:\\d{2}(?:\\.\\d*)?)(?:(?:-(?:\\d{2}):(?:\\d{2})|Z)?)`
+
+        // TODO : handle more date formats
 
-        return Number.parseInt(str, 10)
+        return `\\b(?<date${index}>(${dateRegex})|(${isoDateRegex})|(${isoDatetimeRegex}))\\b`
+    }
+
+    public getValue(str: string): Date {
+        // TODO : handle invalid dates?
+        return new Date(str)
```

No, if date is invalid we've nothing to do.
It's due to value in feature file.

---

I am not really sure about this. I mean, having to check invalid date so it fails early and ensures that only valid dates are accepted. But on the same time, some devs may need to check for invalid date themselves.

I would opt for option 1 but option 2 seems also a good idea. Need to check how other cucumber libraries deal with it.

---

In .NET, we get this kind of error: `System.FormatException : String '31/31/2019' was not recognized as a valid DateTime.`

This is logic because .NET does not hide errors while in JS... I am more convinced that throwing an error is better here because any developer using this library will assume that if a `Date` param is consumed, it must be valid.

---

Ok, we can check if date equals `'Invalid Date'` and throw an error.

An Error extends of `VitestsCucumberError`.